### PR TITLE
Give the project type "go" higher precedence than universal types

### PR DESCRIPTION
**Pull request by knu**
It is so common for a Go-based project to have a Makefile.

-----------------

Before submitting a PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x] The commits are consistent with our [contribution guidelines](../blob/master/CONTRIBUTING.md)
- [ ] You've added tests (if possible) to cover your change(s)
- [ ] All tests are passing ([`eldev test`](https://github.com/doublep/eldev))
- [x] The new code is not generating bytecode or `M-x checkdoc` warnings
- [x] You've updated the [changelog](../blob/master/CHANGELOG.md) (if adding/changing user-visible functionality)
- [ ] You've updated the readme (if adding/changing user-visible functionality)

### CODE REVIEW

**Comment by knu** [comment]
Here's a workaround you can use for now:

```elisp
(setq projectile-project-types
      (sort projectile-project-types
            #'(lambda (a b) (memq (car b) '(make gnumake cmake)))))
```

**Comment by knu** [comment]
A quick survey finds that so many popular Go projects have a Makefile.

- https://github.com/kubernetes/kubernetes
- https://github.com/docker/compose
- https://github.com/hashicorp/packer
- https://github.com/prometheus/prometheus
- https://github.com/hashicorp/terraform
- https://github.com/labstack/echo
- https://github.com/gin-gonic/gin
- https://github.com/junegunn/fzf
…

File: projectile.el [hunk 1]
```diff
@@ -3128,6 +3124,11 @@ a manual COMMAND-TYPE command is created with
                                   :test #'projectile--cmake-test-command
                                   :install #'projectile--cmake-install-command
                                   :package "cmake --build build --target package")
+;; Go
```
Comment (bbatsov) [inline projectile.el:3127]: Might be good to add a comment this should have higher priority than Make.

Comment (knu) [reply projectile.el]: Thanks for pointing that out.  Done!

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**